src/main/java/org/sonatype/gradle/plugins/scan/common/DependenciesFinder.java

```diff
@@ -87,39 +90,58 @@ public class DependenciesFinder
   public Set<ResolvedDependency> findResolvedDependencies(
       Project project,
       boolean allConfigurations,
-      Map<String, String> variantAttributes)
+      Map<String, String> variantAttributes,
+      boolean excludeCompileOnlyDependencies)
   {
+    Set<String> compileOnlyDependenciesIds =
+        excludeCompileOnlyDependencies ? getCompileOnlyDependencyIds(project) : Collections.emptySet();
+
     return new LinkedHashSet<>(new LinkedHashSet<>(project.getConfigurations()).stream()
         .filter(configuration -> isAcceptableConfiguration(configuration, allConfigurations))
-        .flatMap(configuration -> getDependencies(project, configuration, variantAttributes,
-            resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream()))
+        .flatMap(configuration -> {
+          Stream<ResolvedDependency> dependencies = getDependencies(project, configuration, variantAttributes,
+              resolvedConfiguration -> resolvedConfiguration.getFirstLevelModuleDependencies().stream());
+
+          if (!compileOnlyDependenciesIds.isEmpty() && shouldRemoveCompileOnlyDependencies(project, configuration)) {
```

Comment:
Question. Why is `shouldRemoveCompileOnlyDependencies` needed?

Reply:
> Question. Why is `shouldRemoveCompileOnlyDependencies` needed?

![image](https://github.com/sonatype-nexus-community/scan-gradle-plugin/assets/10326286/8e5e2f77-b994-4215-a225-119870e2757c)
Taken from https://medium.com/agorapulse-stories/gradle-configurations-explained-4b9608dd5e35

From the image above we see the `compileClasspath` configuration containing dependencies from `compileOnly` and `implementation` while `runtimeClasspath` does it from `implementation` and `runtimeOnly`

Let's say a project has the library `bar:1.0.0` as transitive dependency from `foo:2.0.0` in the `compileOnly` configuration and also `bar:1.0.0` is declared as direct dependency in `implementation` (similar to what's intented to be done on this plugin with `jgit`).

Now, when that code is iterating over the `compileClasspath` configuration, `shouldRemoveCompileOnlyDependencies` is the method with the logic to determine that the current configuration extends from `compileOnly` and thefore `foo:2.0.0` and `bar:1.0.0` should be removed.

Later, when iterating over the `runtimeClasspath` configuration, since `shouldRemoveCompileOnlyDependencies` would tell that such configuration doesn't extend from `compileOnly` then we don't remove any dependencies from it so `bar:1.0.0` remains in that list. Which is the expected behavior.

An edge case if you will, but we'd be covered.